core/unit_test/TestMathematicalFunctions.hpp

```diff
@@ -1791,6 +1795,121 @@ struct TestIsNaN {
 TEST(TEST_CATEGORY, mathematical_functions_isnan) {
   TestIsNaN<TEST_EXECSPACE>();
 }
+
+template <class Space, class FP16Type>
+struct TestNextAfter {
```

Comment:
If we actually need a separate test implementation rename to `TestNextAfterHalf` or so.